drivers/SmartThings/zigbee-button/src/aqara/init.lua

```diff
@@ -0,0 +1,108 @@
+-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
+
+local battery_defaults = require "st.zigbee.defaults.battery_defaults"
+local clusters = require "st.zigbee.zcl.clusters"
+local cluster_base = require "st.zigbee.cluster_base"
+local data_types = require "st.zigbee.data_types"
+local capabilities = require "st.capabilities"
+
+
+local PowerConfiguration = clusters.PowerConfiguration
+local PRIVATE_CLUSTER_ID = 0xFCC0
+local PRIVATE_ATTRIBUTE_ID = 0x0009
+local MFG_CODE = 0x115F
+
+local MULTISTATE_INPUT_CLUSTER_ID = 0x0012
+local PRESENT_ATTRIBUTE_ID = 0x0055
+
+local FINGERPRINTS = {
+  { mfr = "LUMI", model = "lumi.remote.b1acn02" }
+}
+
+local configuration = {
+    {
+      cluster = MULTISTATE_INPUT_CLUSTER_ID,
+      attribute = PRESENT_ATTRIBUTE_ID,
+      minimum_interval = 3,
+      maximum_interval = 7200,
+      data_type = data_types.Uint16,
+      reportable_change = 1
+    },
+    {
+      cluster = PowerConfiguration.ID,
+      attribute = PowerConfiguration.attributes.BatteryVoltage.ID,
+      minimum_interval = 30,
+      maximum_interval = 3600,
+      data_type = PowerConfiguration.attributes.BatteryVoltage.base_type,
+      reportable_change = 1
+    }
+}
+
+local function present_value_attr_handler(driver, device, value, zb_rx)
+    if value.value == 1 then
+        device:emit_event(capabilities.button.button.pushed({state_change = true}))
+    elseif value.value == 2 then
+        device:emit_event(capabilities.button.button.double({state_change = true}))
+    elseif value.value == 0 then
+        device:emit_event(capabilities.button.button.held({state_change = true}))
+    end
+end
+
+local is_aqara_products = function(opts, driver, device)
+    for _, fingerprint in ipairs(FINGERPRINTS) do
+      if device:get_manufacturer() == fingerprint.mfr and device:get_model() == fingerprint.model then
+        return true
+      end
+    end
+    return false
+end
+
+local function device_init(driver, device)
+    battery_defaults.build_linear_voltage_init(2.6, 3.0)(driver, device)
+    if configuration ~= nil then
+      for _, attribute in ipairs(configuration) do
+        device:add_configured_attribute(attribute)
+        device:add_monitored_attribute(attribute)
+      end
+    end
+end
+
+  local function added_handler(self, device)
+    device:send(cluster_base.write_manufacturer_specific_attribute(device,
+                PRIVATE_CLUSTER_ID, PRIVATE_ATTRIBUTE_ID, MFG_CODE, data_types.Uint8, 1))
+    device:emit_event(capabilities.button.supportedButtonValues({"pushed","held","double"}, {visibility = { displayed = false }}))
+    device:emit_event(capabilities.button.numberOfButtons({value = 1}))
+    device:emit_event(capabilities.button.button.pushed({state_change = false}))
+    device:emit_event(capabilities.battery.battery(100))
```

Comment:
I could be missing something, but is the battery value held at 100%? Is there a specific function that is updating the value? Or is it an automatic reportable change?

Reply:
This is the "added" lifecycle event handler that fires when the device is added for the first time. In general we do this so that the value is populated on add so there are no UI hiccups.

After the fact, the value will be read using the standard zigbee attributes and handled by the defaults.